src/workerd/util/autogate.c++

```diff
@@ -53,19 +57,41 @@ bool Autogate::isEnabled(AutogateKey key) {
   KJ_IF_SOME(a, globalAutogate) {
     return a.gates[(unsigned long)key];
   }
-  LOG_ERROR_PERIODICALLY(
-      kj::str("Autogates not initialised, check for ", key, " will have no effect"));
-  return false;
+
+  if (getenv("WORKERD_ALL_AUTOGATES") != nullptr) {
+    Autogate::initAllAutogates();
```

Comment:
Why do you need to init them at all?

Why not just do:

```
static const bool defaultResult = getenv("WORKERD_ALL_AUTOGATES") != nullptr;
return defaultResult;
```

Seems a lot simpler, and avoids thread-safety issues.

Reply:
This is really nice. You're absolutely right. Thank you!